azurerm/resource_arm_sql_firewall_rule.go

```diff
@@ -51,96 +50,86 @@ func resourceArmSqlFirewallRule() *schema.Resource {
 	}
 }
 
-func resourceArmSqlFirewallRuleCreate(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
-
-	createRequest := rivieraClient.NewRequest()
-	createRequest.Command = &sql.CreateOrUpdateFirewallRule{
-		Name:              d.Get("name").(string),
-		ResourceGroupName: d.Get("resource_group_name").(string),
-		ServerName:        d.Get("server_name").(string),
-		StartIPAddress:    azure.String(d.Get("start_ip_address").(string)),
-		EndIPAddress:      azure.String(d.Get("end_ip_address").(string)),
-	}
+func resourceArmSqlFirewallRuleCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	sqlFirewallRulesClient := meta.(*ArmClient).sqlFirewallRulesClient
 
-	createResponse, err := createRequest.Execute()
-	if err != nil {
-		return fmt.Errorf("Error creating SQL Server Firewall Rule: %s", err)
-	}
-	if !createResponse.IsSuccessful() {
-		return fmt.Errorf("Error creating SQL Server Firewall Rule: %s", createResponse.Error)
-	}
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	serverName := d.Get("server_name").(string)
+	startIpAddress := d.Get("start_ip_address").(string)
+	endIpAddress := d.Get("end_ip_address").(string)
 
-	readRequest := rivieraClient.NewRequest()
-	readRequest.Command = &sql.GetFirewallRule{
-		Name:              d.Get("name").(string),
-		ResourceGroupName: d.Get("resource_group_name").(string),
-		ServerName:        d.Get("server_name").(string),
+	props := sql.FirewallRuleProperties{
+		StartIPAddress: &startIpAddress,
+		EndIPAddress:   &endIpAddress,
+	}
```

Comment:
sorry, missed this - [here's an example from the SQL Server Resource](https://github.com/terraform-providers/terraform-provider-azurerm/blob/2c73509a1b276a4439c9da5d722ffd59bb011a9e/azurerm/resource_arm_sql_server.go#L84)